.evergreen/compilation-context-expansions.sh

```diff
@@ -1,4 +1,4 @@
-export DISTRO_UPLOADLIST_REGEX="^(rhel70|win|rhel81-power8|amazon2-arm64|macos)"
+export DISTRO_UPLOADLIST_REGEX="^(rhel70|win|rhel72-zseries|rhel81-power8|amazon2-arm64|macos)"
```

Comment:
This was missed in https://github.com/mongodb-js/mongosh/pull/1596, sorry for not catching it during review, but the e2e tests on s390x won’t run if their compilation task doesn’t upload anything to test :)